plugins/notifications-backend-module-slack/src/lib/SlackNotificationProcessor.ts

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AuthService,
+  CacheService,
+  DiscoveryService,
+  LoggerService,
+} from '@backstage/backend-plugin-api';
+import { CatalogClient } from '@backstage/catalog-client';
+import { Entity, UserEntity, parseEntityRef } from '@backstage/catalog-model';
+import { Config } from '@backstage/config';
+import { Notification } from '@backstage/plugin-notifications-common';
+import {
+  NotificationProcessor,
+  NotificationSendOptions,
+} from '@backstage/plugin-notifications-node';
+import { ANNOTATION_NOTIFICATION_SLACK_CHANNEL_ID } from './constants';
+
+const DEFAULT_CACHE_TTL = 1000 * 60 * 60; // 1 hour
+
+export class SlackNotificationProcessor implements NotificationProcessor {
+  private readonly logger: LoggerService;
+  private readonly directWebhookUrl?: string;
+  private readonly channelWebhookUrl?: string;
+  private readonly cache: CacheService;
+  private readonly ttl: number;
+  private readonly catalogClient: CatalogClient;
+  private readonly auth: AuthService;
+
+  static fromConfig(
+    config: Config,
+    options: {
+      auth: AuthService;
+      discovery: DiscoveryService;
+      logger: LoggerService;
+      cache: CacheService;
+      catalogClient?: CatalogClient; // used for testing
```

Comment:
could also make it required, just always passing it in (moving the responsibility out), but matter of taste